File: treat/frontend/sanity/schemaTypes/agencyMessage.ts
import { defineField, defineType } from 'sanity';

export default defineType({
  name: 'agencyMessage',
  title: 'Agency Message',
  type: 'document',
  fields: [
    defineField({
      name: 'channel',
      title: 'Channel',
      type: 'string',
      description: 'Channel ID: "general" or ticket reference like "UIA-2026-0001"',
      validation: r => r.required(),
    }),
    defineField({
      name: 'content',
      title: 'Content',
      type: 'text',
      validation: r => r.required(),
    }),
    defineField({
      name: 'senderName',
      title: 'Sender Name',
      type: 'string',
      validation: r => r.required(),
    }),
    defineField({
      name: 'senderAgency',
      title: 'Sender Agency',
      type: 'reference',
      to: [{ type: 'agency' }],
    }),
    defineField({
      name: 'senderAgencyCode',
      title: 'Sender Agency Code',
      type: 'string',
    }),
    defineField({
      name: 'senderEmail',
      title: 'Sender Email',
      type: 'email',
    }),
    defineField({
      name: 'attachments',
      title: 'Attachments',
      type: 'array',
      of: [{ type: 'file' }],
    }),
    defineField({
      name: 'isInternal',
      title: 'Internal Only',
      type: 'boolean',
      initialValue: false,
    }),
    defineField({
      name: 'sentAt',
      title: 'Sent At',
      type: 'datetime',
    }),
  ],
  preview: {
    select: { title: 'senderName', subtitle: 'channel' },
  },
});

File: backend/src/OscApi/Models/AgencyMessage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OscApi.Models;

[Table("agency_messages")]
public class AgencyMessage
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required, MaxLength(100)]
    public string Channel { get; set; } = string.Empty;

    [Required, MaxLength(5000)]
    public string Content { get; set; } = string.Empty;

    [Required, MaxLength(100)]
    public string SenderName { get; set; } = string.Empty;

    [MaxLength(20)]
    public string? SenderAgencyCode { get; set; }

    [MaxLength(255)]
    public string? SenderEmail { get; set; }

    public bool IsInternal { get; set; }

    public DateTimeOffset SentAt { get; set; } = DateTimeOffset.UtcNow;
}
